ignite/pkg/cosmosclient/rpc.go

```diff
@@ -0,0 +1,233 @@
+package cosmosclient
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/tendermint/tendermint/libs/bytes"
+	"github.com/tendermint/tendermint/rpc/client"
+	rpcclient "github.com/tendermint/tendermint/rpc/client"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	"github.com/tendermint/tendermint/types"
+)
+
+// rpcWrapper is a rpclient.Client but with more contextualized errors.
+// Useful because the original implementation may return JSON errors when the
+// requested node is busy, which is confusing for the user. With rpcWrapper,
+// the error is prefixed with 'error while requesting node xxx: JSON error'.
+type rpcWrapper struct {
+	rpcclient.Client
+	nodeAddress string
+}
```

Comment:
Maybe we should leave a TODO with a link to the [Tendermint issue](https://github.com/tendermint/tendermint/issues/9312) so we can remove the wrapper once the issue is fixed, if I understand correctly.

Reply:
done [744a2b7](https://github.com/ignite/cli/pull/2813/commits/744a2b723ca2cd6c4065a054abf991b75f8002fd)